### PR TITLE
chore: drop tox

### PR DESCRIPTION
**Pull request by neersighted**
Drop tox as we have not made use of it in the development of Poetry for a long time.

Additionally, switch to the new `tool.poetry.group` syntax.

### CODE REVIEW

**Comment by Secrus** [comment]
I am not sure about dropping tox. I has its use in local testing on multiple versions.

File: pyproject.toml [hunk 1]
```diff
@@ -71,8 +71,7 @@ xattr = { version = "^0.9.7", markers = "sys_platform == 'darwin'" }
 urllib3 = "^1.26.0"
 dulwich = "^0.20.44"
 
-[tool.poetry.dev-dependencies]
-tox = "^3.18"
+[tool.poetry.group.test.dependencies]
```
Comment (Secrus) [inline pyproject.toml:74]: I don't like just changing labels like this. If we want to make use of groups, we should split the deps properly (mypy and pre-commit are not "test" deps)

**Review by Secrus — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (neersighted) [reply pyproject.toml:74]: That was a brainfart -- it was supposed to be dev

**Comment by neersighted** [comment]
> I am not sure about dropping tox. I has its use in local testing on multiple versions.

Personally I find it to be of relatively low value as the matrix is wide with OS versions, so you really need to rely on CI to test widely. If we really are going to make it part of the repo tooling we need to maintain it, document it, and make it the default option for contributors -- multiple methods to run tests locally is confusing at best and harmful at worst.

**Review by Secrus — APPROVED** [review]

**Pull request merged**